lib/cartopy/io/img_tiles.py

```diff
@@ -497,6 +497,50 @@ def find_images(self, target_domain, target_z, start_tile=None):
                 yield self.tms_to_quadkey(tile, google=True)
 
 
+class OrdnanceSurvey(GoogleWTS):
+    # https://developer.ordnancesurvey.co.uk/os-api-framework-agreement
+    # https://apidocs.os.uk/docs/os-maps-wmts
+
+    def __init__(self, apikey, layer='Road', style=True,
+                 desired_tile_form='RGB'):
+        """
+        Parameters
+        ----------
+        apikey: required
+            The authentication key provided by OS to query the maps API
+        layer: optional
+            The style of the OS map tiles. One of 'Outdoor', 'Road',
+            'Light', 'Night', 'Leisure'. Defaults to 'Road'.
+        style: optional
+            Indicates if the map should be styled. Defaults to True.
+        desired_tile_form: optional
+            Defaults to 'RGB'.
+        """
+        super(OrdnanceSurvey, self).__init__(
+            desired_tile_form=desired_tile_form)
+        self.apikey = apikey
+        self.style = 'true' if style else 'false'
+
+        if layer not in {'Outdoor', 'Road', 'Light', 'Night', 'Leisure'}:
```

Comment:
You've set up the options here as a dictionary containing keys only. Better would be a list, or even a tuple, containing these values.

👍 for checking (and failing) early though.

Reply:
This is a set, which clarifies the uniqueness of the contents.

Reply:
I always forget about that syntax! Nevertheless, I think a list would suffice here.